src/MainWindow.vala

```diff
@@ -1344,9 +1344,77 @@ namespace Terminal {
             return;
         }
 
+        private void zoom_changed (TerminalWidget terminal_widget) {
+            save_opened_terminal_zoom (terminal_widget);
+        }
+
+        private void cwd_changed () {
+            check_for_tabs_with_same_name ();
```

Comment:
I don't think this call is necessary - the function is called by the `window_title_changed` signal handler anyway.

Reply:
I am sorry, I do not know. Eventually, I would remove call `check_for_tabs_with_same_name` in master branch as this should be focused only on continuous saving in case this branch would be reverted for example in the future.